# Libraries

In [15]:
import pandas as pd

from thefuzz import process

# Configurations

In [16]:
# Pandas configuration for long phrases
pd.set_option('display.max_colwidth', 50)

# Data

In [17]:
df = pd.read_csv('data/Jobs.csv')

In [18]:
df.head()

Unnamed: 0                                       title  \
0           0  Senior Analyst, Data Science and Analytics   
1           1                       Senior Data Scientist   
2           2                   Lead Data Science Analyst   
3           3                         Data Science Intern   
4           4                              Data Scientist   

                             company  announcement  \
0                         TransUnion      The Muse   
1             Grubhub Holdings, Inc.  ZipRecruiter   
2        Discover Financial Services      LinkedIn   
3                      AbelsonTaylor  Startup Jobs   
4  NORC at the University of Chicago   SimplyHired   

                                         description  
0  TransUnion's Job Applicant Privacy Notice  Wha...  
1  About The Opportunity  We're all about connect...  
2  Discover. A brighter future.  With us, you’ll ...  
3  Are you a 2023 college graduate or rising coll...  
4  JOB DESCRIPTION:  At NORC, Data Scientists pla...

# Cleaning Data

In [19]:
df = df.drop('Unnamed: 0', axis=1)

In [20]:
df.head()

title  \
0  Senior Analyst, Data Science and Analytics   
1                       Senior Data Scientist   
2                   Lead Data Science Analyst   
3                         Data Science Intern   
4                              Data Scientist   

                             company  announcement  \
0                         TransUnion      The Muse   
1             Grubhub Holdings, Inc.  ZipRecruiter   
2        Discover Financial Services      LinkedIn   
3                      AbelsonTaylor  Startup Jobs   
4  NORC at the University of Chicago   SimplyHired   

                                         description  
0  TransUnion's Job Applicant Privacy Notice  Wha...  
1  About The Opportunity  We're all about connect...  
2  Discover. A brighter future.  With us, you’ll ...  
3  Are you a 2023 college graduate or rising coll...  
4  JOB DESCRIPTION:  At NORC, Data Scientists pla...

In [21]:
# Data types of columns
df.dtypes

title           object
company         object
announcement    object
description     object
dtype: object

In [22]:
# Dataframe information
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 790 entries, 0 to 789
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   title         790 non-null    object
 1   company       790 non-null    object
 2   announcement  790 non-null    object
 3   description   790 non-null    object
dtypes: object(4)
memory usage: 24.8+ KB


In [23]:
# Get duplicated values
df[df.duplicated()]

Empty DataFrame
Columns: [title, company, announcement, description]
Index: []

In [24]:
# Missing values
df[df['description'].isna()]

# There is no NA values. 

Empty DataFrame
Columns: [title, company, announcement, description]
Index: []

In [25]:
# All type of unique titles of the job
df['title'].nunique()

515

There are 515 unique job title, but the most are repeated, so we have to classify this titles in some unique occurences. 

# Create some unique categories for job title

In [ ]:
job_description = ['Senior Analyst', 'Senior Data Scientist', 'Data Science Manager', ]